pkg/k8scontext/context.go

```diff
@@ -167,7 +173,12 @@ func (c *Context) Run(stopChannel chan struct{}, omitCRDs bool, envVariables env
 
 	glog.V(1).Infoln("Waiting for initial cache sync")
 	if !cache.WaitForCacheSync(stopChannel, hasSynced...) {
-		return ErrorFailedInitialCacheSync
+		e := controllererrors.NewError(
```

Comment:
Give how the NewError is being used, the message is kind of duplicated with the code, the code is also a string underneath? we could just keep the Code field for NewError

Reply:
for now, I have added human readable messages for each of the errors.